### ISSUE TITLE
Radarr-SMA ffmpeg binary not found

### ISSUE THREAD

**Issue by CoderKiwi** [mdhiggins/sickbeard_mp4_automator#926]
Radarr shows the following log errors:

```
  | python3 | converter.ffmpeg.FFMpegError: ffmpeg binary not found: /usr/bin/ffmpeg | 9:49pm
  | python3 | raise FFMpegError("ffmpeg binary not found: " + self.ffmpeg_path) | 9:49pm
  | python3 | File "/usr/local/bin/sma/sickbeard_mp4_automator/converter/ffmpeg.py", line 364, in __init__ | 9:49pm
  | python3 | ffprobe_path=ffprobe_path) | 9:49pm
  | python3 | File "/usr/local/bin/sma/sickbeard_mp4_automator/converter/__init__.py", line 27, in __init__ | 9:49pm
  | python3 | info = Converter(self.FFMPEG_PATH, self.FFPROBE_PATH).probe(inputfile) | 9:49pm
  | python3 | File "/usr/local/bin/sma/sickbeard_mp4_automator/mkvtomp4.py", line 302, in generateOptions | 9:49pm
  | python3 | options = self.generateOptions(inputfile, original=original) | 9:49pm
  | python3 | File "/usr/local/bin/sma/sickbeard_mp4_automator/mkvtomp4.py", line 189, in process | 9:49pm
  | python3 | output = converter.process(inputfile, original=original) | 9:49pm
  | python3 | File "/usr/local/bin/sma/sickbeard_mp4_automator/postRadarr.py", line 45, in <module> | 9:49pm
  | python3 | Traceback (most recent call last): | 9:49pm
```

On the [docker hub page](https://hub.docker.com/r/mdhiggins/radarr-sma/), it says the ffmpeg and ffprobe libraries are accessible at `/usr/local/bin/ffmpeg/ffmpeg` and `/usr/local/bin/ffmpeg/ffprobe` respectively. So I edited `autoProcess.ini` to use those paths. 

However the logs above show `postRadarr.py` is looking for libraries are `/usr/bin/...` instead. How can I resolve this?

**Comment by mdhiggins** [comment]
Can you post your autoProcess.ini file as well as your docker configuration?

Either your autoProcess.ini file is incorrectly configured or your didn't put it in the right spot and its using the default settings

Defaults settings only specify ffmpeg and ffprobe executable names without a path which on linux would resolve to things in /usr/bin

**Comment by CoderKiwi** [comment]
That's exactly what I think is happening too; just not sure why it's doing that. 

**docker-compose.yml** where "blah" is censoring:

```
  radarr:
    image: mdhiggins/radarr-sma:latest
    container_name: radarr
    privileged: false
    restart: always
    depends_on:
      - blah
      - blah
      - blah
    ports:
      - "blah:7878"
    volumes:
      - ./radarr:/config
      - ./radarr/sma/autoProcess.ini:/usr/local/bin/sma/sickbeard_mp4_automator/autoProcess.ini:ro
      - ../Shared Media:/Shared Media
      - ./blah/complete:/data/complete # from blah
      - ./blah/Downloads/complete:/config/Downloads/complete # from blah
    environment:
      TZ: blah
      PUID: 1000
      PGID: 1000
```

**autoProcess.ini**:

```
[Radarr]
host = localhost
port = blah
web_root = /radarr
ssl = False
apikey = blah

[MP4]
ffmpeg = /usr/local/bin/ffmpeg/ffmpeg
ffprobe = /usr/local/bin/ffmpeg/ffprobe
threads = auto
output_directory =
copy_to =
move_to =
output_extension = mp4
output_format = mp4
delete_original = True
relocate_moov = True
video-codec = h264,x264
video-bitrate =
video-crf =
video-max-width =
video-profile =
h264-max-level =
use-qsv-decoder-with-encoder = True
use-hevc-qsv-decoder = False
enable_dxva2_gpu_decode = False
ios-audio = True
ios-first-track-only = False
ios-audio-filter =
ios-move-last = False
max-audio-channels =
audio-codec = ac3
audio-language = eng
audio-default-language = eng
audio-channel-bitrate =
audio-filter =
audio-copy-original = False
audio-first-track-of-language = False
subtitle-codec = mov_text
subtitle-language = eng
subtitle-default-language = eng
subtitle-encoding =
fullpathguess = True
convert-mp4 = False
tagfile = True
tag-language = en
download-artwork = Poster
download-subs = False
embed-subs = True
embed-only-internal-subs = False
sub-providers = addic7ed,podnapisi,thesubdb,opensubtitles
permissions = 0777
post-process = False
pix-fmt =
aac_adtstoasc = False
postopts =
preopts =
```

**Comment by mdhiggins** [comment]
At first glance everything looks ok. That being said, I do have a couple things you could try

I would try switching your volume paths to use the absolute path instead of the relative path (no ./, use /home/usr/radarr or whatever your actual path is).

Another thing is to verify the permissions of your autoProcess.ini file, make sure its owned by whoever your user 1000 group 1000 is so that it can be accessed from the user within the docker container

Another change is that it shouldn’t be read only, so you can drop the :ro (though that’s just an extra and not related to your current issue). The script occasionally will get new options and add those to the config and without write permission will crash when attempting to do so.

I would give those changes a try and see if that makes any difference

**Comment by CoderKiwi** [comment]
I’ll give those a go, thanks!

**Comment by CoderKiwi** [comment]
Seems to be working! Very neat -- I've been wanting to use these scripts for a while. Thank you. 

I did the following like you suggested:
- Removed :ro (optional)
- Checked permissions (seemed fine), chown'd it anyways
- Changed to absolute file paths

I think it was the absolute file paths that did it. 

One quick question: now that media is being converted going forward as things are downloaded, what's a way I can batch convert all my existing media? Can I set the `manual.py` script loose on, say, a movies directory? What command would I use for that?
